src/Lifecycle.ts

```diff
@@ -1118,8 +1122,12 @@ async function clearStorage(opts?: { deleteEverything?: boolean }): Promise<void
             logger.error("idbDelete failed for account:mx_access_token", e);
         }
 
-        // now restore those invites and registration time
+        // now restore those invites, registration time and previously set language
         if (!opts?.deleteEverything) {
+            if (language) {
```

Comment:
this if check is redundant, language will always be set due to no excludeDefault